### PR TITLE
add unit testing on project

### PR DESCRIPTION
**Pull request by AlexeyGrishechkin**
fix bug with showing total page
improve helpers function

### CODE REVIEW

File: src/utils/helpers/createGenreList.js [hunk 1]
```diff
@@ -1,6 +1,11 @@
 export const createGenreList = (genreList, filmGenres) => {
-  return genreList
-    .filter((genre) => filmGenres.indexOf(genre.id) !== -1)
-    .map((genre) => genre.name)
-    .join(", ");
+  const findGenresInGenresList = genreList.filter(
```
Comment (alexanderchernetsky) [inline src/utils/helpers/createGenreList.js]: incorrect naming. It's not a function to be started from the verb "find". Example of correct name - genresFromGenresList

File: src/utils/helpers/describeRating.js [hunk 2]
```diff
@@ -1,13 +1,19 @@
 export const describeRating = (rating) => {
-  if (rating === 0 || !rating) {
+  const convertRatingToNumber = Number(rating);
```
Comment (alexanderchernetsky) [inline src/utils/helpers/describeRating.js]: Incorrect naming. Can be ratingConvertedToNumber . As it's a number it should start from a noun.

File: src/utils/helpers/describeRating.js [hunk 3]
```diff
@@ -1,13 +1,19 @@
...
+
+  if (
+    convertRatingToNumber === 0 ||
```
Comment (alexanderchernetsky) [inline src/utils/helpers/describeRating.js]: convertRatingToNumber === 0 || !convertRatingToNumber - can be simplified to just !convertRatingToNumber

File: src/utils/helpers/expandList.js [hunk 4]
```diff
@@ -1,2 +1,9 @@
-export const expandList = (list = []) => list.map((item) => item.name).join(", ");
+export const expandList = (list = []) => {
```
Comment (alexanderchernetsky) [inline src/utils/helpers/expandList.js]: Bad name for the function. In fact it's building a string from the array. Expanding/collapsing is a different functionality. Please think of a better name.

File: src/utils/helpers/expandList.js [hunk 5]
```diff
@@ -1,2 +1,9 @@
...
+  const filterList = list.filter((item) => item.name);
 
+  if (list.length === 0 || filterList.length === 0) {
```
Comment (alexanderchernetsky) [inline src/utils/helpers/expandList.js]: No need to check list.length. filterList.length is what we are interested in.

File: src/utils/helpers/tests/checkData.spec.js [hunk 6]
```diff
@@ -0,0 +1,13 @@
+import { checkData } from "../checkData";
+
+describe("checkData gets two parameters and returns these joined parameters as string", () => {
```
Comment (alexanderchernetsky) [inline src/utils/helpers/tests/checkData.spec.js]: In describe you should leave only function name. Please correct this in all tests

File: src/utils/helpers/tests/checkData.spec.js [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+  test("get data and description", () => {
+    expect(checkData(30000, "$")).toBe("30000 $");
+  });
+  test("get only one parameter and return string", () => {
```
Comment (alexanderchernetsky) [inline src/utils/helpers/tests/checkData.spec.js]: I don't like the wording. It's much clearer to say: "given only 1 parameter - returns string containing this parameter"

File: src/utils/helpers/tests/createGenreList.spec.js [hunk 8]
```diff
@@ -0,0 +1,24 @@
+import { createGenreList } from "../createGenreList";
+
+const genreList = [
```
Comment (alexanderchernetsky) [inline src/utils/helpers/tests/createGenreList.spec.js]: As it's a mock data it's name should start from "mock" by convention. E.g. mockGenreList.

File: src/utils/helpers/tests/createGenreList.spec.js [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+  { id: 1, name: "action" },
+  { id: 2, name: "comedy" },
+  { id: 3, name: "thriller" },
+  { id: 4, name: "cartoon" },
+  { id: 5, name: "family" },
+];
+const filmGenre = [1, 3];
+
+describe("createGenreList get genreList array of objects with id and name and filmGenre array of number (genres id) and returns names of genre as string", () => {
```
Comment (alexanderchernetsky) [inline src/utils/helpers/tests/createGenreList.spec.js]: Only "createGenreList" should be in the upper describe. Please correct the same mistake in all places.

File: src/utils/helpers/tests/describeRating.spec.js [hunk 10]
```diff
@@ -0,0 +1,16 @@
+import { describeRating } from "../describeRating";
+
+describe("describeRating gets a number and returns string with rate", () => {
```
Comment (alexanderchernetsky) [inline src/utils/helpers/tests/describeRating.spec.js]: In the upper "describe" you should write which function you're testing. In "it" you should write what the test case you're testing. E.g. 
describe("describeRating", () => {
  it('given a rating number - returns estimation string') ...

File: src/utils/helpers/tests/expandList.spec.js [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+describe("expandList gets array of objects with key name and returns list of name as string", () => {
+  test("get valid array of objects with key name", () => {
+    expect(expandList(countriesList)).toBe(
+      "Canada, United Kingdom, Denmark, Belgium"
+    );
+  });
+  test("get array with different parameters, find key name and return list of name as string", () => {
+    expect(expandList(invalidCountriesList)).toBe("Canada, Denmark");
+  });
+  test("get nothing and return string no data", () => {
```
Comment (alexanderchernetsky) [inline src/utils/helpers/tests/expandList.spec.js]: Replace "get" with "given". It's incorrect from the English language perspective. Word "get" has a meaning - получить. In our tests "given" suits better.

**Review by alexanderchernetsky — CHANGES_REQUESTED** [review]

File: src/utils/helpers/tests/expandList.spec.js (hunk 11 above)
Comment (AlexeyGrishechkin) [reply src/utils/helpers/tests/expandList.spec.js]: fixed

File: src/utils/helpers/tests/describeRating.spec.js (hunk 10 above)
Comment (AlexeyGrishechkin) [reply src/utils/helpers/tests/describeRating.spec.js]: fixed

File: src/utils/helpers/tests/createGenreList.spec.js (hunk 9 above)
Comment (AlexeyGrishechkin) [reply src/utils/helpers/tests/createGenreList.spec.js]: fixed

File: src/utils/helpers/tests/createGenreList.spec.js (hunk 8 above)
Comment (AlexeyGrishechkin) [reply src/utils/helpers/tests/createGenreList.spec.js]: fixed

File: src/utils/helpers/tests/checkData.spec.js (hunk 7 above)
Comment (AlexeyGrishechkin) [reply src/utils/helpers/tests/checkData.spec.js]: fixed

File: src/utils/helpers/tests/checkData.spec.js (hunk 6 above)
Comment (AlexeyGrishechkin) [reply src/utils/helpers/tests/checkData.spec.js]: fixed

File: src/utils/helpers/expandList.js (hunk 5 above)
Comment (AlexeyGrishechkin) [reply src/utils/helpers/expandList.js]: fixed

File: src/utils/helpers/expandList.js (hunk 4 above)
Comment (AlexeyGrishechkin) [reply src/utils/helpers/expandList.js]: fixed

File: src/utils/helpers/describeRating.js (hunk 3 above)
Comment (AlexeyGrishechkin) [reply src/utils/helpers/describeRating.js]: fixed

File: src/utils/helpers/describeRating.js (hunk 2 above)
Comment (AlexeyGrishechkin) [reply src/utils/helpers/describeRating.js]: fixed

File: src/utils/helpers/createGenreList.js (hunk 1 above)
Comment (AlexeyGrishechkin) [reply src/utils/helpers/createGenreList.js]: fixed

**Pull request merged**